### ISSUE TITLE
[naming-convention] custom severity for single conventions

### ISSUE THREAD

**Issue by htho** [typescript-eslint/typescript-eslint#3035]
I am not sure where to ask/request this.

Right now rules are defined as an array: [("error"|"warn"), Convention, Convention, ...]

I'd like to define some conventions as errors and some conventions as warnings.

I don't think that this is possible right now. I have no preference on how to configure this.

- [ ] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```ts
// Variables should be camelCase
const this_is_an_error = "foo";
// Enums should be PascalCase
enum thisShouldJustWarn {
	OK=0,
}
```

```JSON
{
  "rules": {
    "@typescript-eslint/naming-convention": ["<setting>"]
  }
}
```

**Expected Result**

* the const is marked as an error
* the enum is marked as a warning

**Actual Result**

everything is either an error or a warning

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.15.0` |
| `@typescript-eslint/parser`        | `4.15.0` |
| `TypeScript`                       | `4.1.3` |
| `ESLint`                           | `7.19.0` |
| `node`                             | `12.10.0` |

**Comment by bradzacher** [comment]
Unfortunately this is not possible. Eslint only supports one level per rule.
